app/scripts/services/validateRegistrant.js

```diff
@@ -4,52 +4,36 @@ angular.module('confRegistrationWebApp')
   .service('validateRegistrant', function validateRegistrant() {
 
     var blockVisibleRuleCheck = function (block, registrant) {
-      var returnValueAND = true;
-      var returnValueOR = false;
       var answers = registrant.answers;
-      var ruleOperand = block.content && block.content !== '' && !angular.isUndefined(block.content.ruleoperand) ? block.content.ruleoperand : 'AND';
+      var ruleOperand = block.content && block.content.ruleoperand ? block.content.ruleoperand : 'AND';
+      var ruleArray = [];
```

Comment:
I feel like you could just use a counter instead of an array that you are only using the length of.